src/config/loading/secret.rs

```diff
@@ -0,0 +1,341 @@
+use std::{collections::HashMap, io::Read};
+
+use futures::executor;
+use indexmap::IndexMap;
+use once_cell::sync::Lazy;
+use regex::{Captures, Regex};
+use serde::{Deserialize, Serialize};
+use tokio::{io::AsyncWriteExt, process::Command, time::Duration};
+use toml::value::Table;
+use typetag::serde;
+
+use super::{loader, prepare_input};
+use crate::config::{
+    loading::{deserialize_table, ComponentHint, Process},
+    ComponentKey,
+};
+use crate::signal;
+
+static COLLECTOR: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"SECRET\[([[:word:]]+)\.([[:word:].]+)\]").unwrap());
```

Comment:
Nit: might be nice to add some examples of what this matches as a comment since this regex is a bit complex.